### PR TITLE
AB#7290 Switch Discover SDK from a JS file to the NPM package

### PR DESCRIPTION
**Pull request by maxwmaher**
1) Preview Search and Full Page Search are using the 0.2.1 version of the scd-sdk from the private registry.
2) Also includes fixes for user stories 7316, 7318, 7319, & 7320

### CODE REVIEW

File: Website/src/rendering/src/services/DiscoverService.tsx [hunk 1]
```diff
@@ -1,135 +0,0 @@
...
-            preRender: true,
-            properties: {
-              initial: {
-                redirectUrl: '/shop/products?q=',
-                inputQuerySelector: '#search-input',
-              },
-            },
-          },
-        });
-        window.RFK.setWidget('ps_trending_categories', {
```
Comment (jflheureux) [inline Website/src/rendering/src/services/DiscoverService.tsx:120]: Did you forget to setup this widget in the new implementation?

File: Website/src/rendering/src/services/DiscoverService.tsx [hunk 2]
```diff
@@ -1,135 +0,0 @@
...
-    <Script
-      src="/rfk-sdk.js"
-      onLoad={() => {
-        window.RFK.setCredentials({
-          env: 'prod',
-          customerKey: `${DISCOVER_CUSTOMER_KEY}`,
-          apiKey: `${DISCOVER_API_KEY}`,
-          useToken: true,
-        });
-        window.RFK.setWidget('rfkid_11', {
```
Comment (jflheureux) [inline Website/src/rendering/src/services/DiscoverService.tsx:97]: Did you forget to setup this widget in the new implementation?

File: Website/src/rendering/.npmrc [hunk 3]
```diff
@@ -0,0 +1 @@
+@sitecore-discover:registry=https://sitecore.myget.org/F/sc-npm-packages/npm/
```
[the remark this answers is not in the document]
Comment (jflheureux) [reply Website/src/rendering/.npmrc:1]: Removing the file will break the CI build. We will have to find another way to add this registry to the CI build

**Review by jflheureux — CHANGES_REQUESTED** [review]
Quick review. I have not reviewed the component files yet.

File: Website/src/rendering/tsconfig.json [hunk 4]
```diff
@@ -30,13 +38,14 @@
     "allowSyntheticDefaultImports": true,
     "noImplicitReturns": true,
     "noUnusedLocals": true,
-    "noUnusedParameters": true
+    "noUnusedParameters": true,
+    "incremental": true
```
[the remark this answers is not in the document]
Comment (maxwmaher) [reply Website/src/rendering/tsconfig.json:42]: I can't, it looks like this file automatically got adjusted during my first commit (557f95 - Connection to scd-sdk). Do you know what processes affect the change of values in this file?

**Review by jflheureux — APPROVED** [review]

**Pull request merged**